src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -35,17 +49,89 @@ public LibGpiodDriver()
+        }
+
+        private Task InitializeEventDetectionTask(LibGpiodDriverEventHandler eventHandler, CancellationToken token)
+        {
+            return Task.Run(() =>
+            {       
+                while (!(token.IsCancellationRequested || _disposing))
+                {   // WaitEventResult can be TimedOut, EventOccured or Error, in case of TimedOut will continue waiting
+                    WaitEventResult waitResult = Interop.WaitForEventOnLine(eventHandler.PinHandle);
+                    if (waitResult == WaitEventResult.Error)
+                    {
+                        throw ExceptionHelper.GetIOException(ExceptionResource.EventWaitError, Marshal.GetLastWin32Error(), eventHandler.PinNumber);
+                    }
+
+                    if (waitResult == WaitEventResult.EventOccured)
+                    {
+                        int readResult = Interop.ReadEventForLine(eventHandler.PinHandle);
+                        if (readResult == -1)
+                        {
+                            throw ExceptionHelper.GetIOException(ExceptionResource.EventReadError, Marshal.GetLastWin32Error());
+                        }
+
+                        PinEventTypes eventType = (readResult == 1) ? PinEventTypes.Rising : PinEventTypes.Falling;
+                        eventHandler?.OnPinValueChanged(new PinValueChangedEventArgs(eventType, eventHandler.PinNumber), eventType);
+                    }
+                }
+            }, token);
+        }
+
         protected internal override void AddCallbackForPinValueChangedEvent(int pinNumber, PinEventTypes eventType, PinChangeEventHandler callback)
         {
-            throw new NotImplementedException();
+            if (!_pinNumberToEventHandler.TryGetValue(pinNumber, out LibGpiodDriverEventHandler eventHandler))
+            {
+                eventHandler = new LibGpiodDriverEventHandler(pinNumber, new CancellationTokenSource());
+                if (_pinNumberToEventHandler.TryAdd(pinNumber, eventHandler))
+                {
+                    eventHandler.PinHandle = SubscribeForEvent(pinNumber, eventType);
+                    eventHandler.Task = InitializeEventDetectionTask(eventHandler, eventHandler.CancellationTokenSource.Token);
+                }
+            }
+
+            if (eventType.HasFlag(PinEventTypes.Rising))
+            {
+                eventHandler.ValueRising += callback;
+            }
+            else if (eventType.HasFlag(PinEventTypes.Falling))
+            {
+                eventHandler.ValueFalling += callback;
+            }
         }
         
         protected internal override void ClosePin(int pinNumber)
         {
             if (_pinNumberToSafeLineHandle.TryGetValue(pinNumber, out SafeLineHandle pinHandle))
             {
-                Interop.ReleaseGpiodLine(pinHandle);
-                _pinNumberToSafeLineHandle.Remove(pinNumber);
+                //Event listeners might still waiting for event, don't release them 
```

Comment:
nit space after `//`

Reply:
- the comment should be inside the inner `if`
- this could be just a single if with `&&`